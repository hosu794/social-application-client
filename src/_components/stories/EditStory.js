import React, { lazy, Suspense, useEffect } from "react";

import { useSelector, useDispatch } from "react-redux";

import { storyActions } from "../../_actions";

import PropTypes from "prop-types";
import { IStory } from "../../_types";

const EditDashboard = lazy(() => import("./EditDashboard"));

function EditStory(props) {
  const dispatch = useDispatch();

  const currentStory = useSelector((state) => state.stories.currentStory);

  const isLoading = useSelector((state) => state.stories.loading);

  const getStoryById = async () => {
    dispatch(storyActions.getStoryById(props.match.params.id));
  };
  useEffect(() => {
    getStoryById();
  }, []);

  return (
    <Suspense fallback={<div>Loading...</div>}>
      {!isLoading ? <EditDashboard story={currentStory} /> : null}
    </Suspense>
  );
}

EditStory.propTypes = {
  currentStory: PropTypes.objectOf(IStory),
  isLoading: PropTypes.bool.isRequired,
};

export default EditStory;
