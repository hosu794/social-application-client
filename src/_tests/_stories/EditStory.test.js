import React from "react";

import configureStore from "redux-mock-store";

import { MemoryRouter } from "react-router-dom";

import { mount } from "enzyme";
import { Provider } from "react-redux";
import EditStory from "../../_components/stories/EditStory";

const mockStore = configureStore([]);

let store = mockStore({
  stories: {
    loading: false,
    currentStory: {
      id: 1,
      title: "Story Title",
      description: "Story description",
      totalLoves: 12,
      body: "<p>Simple Story</p>",
      createdBy: {
        id: 12,
        name: "Joe Doe",
        username: "exampleusername",
        downloadAvatar: "dsdasdasdasdasdadas.png",
      },
      topic: {
        id: 12,
        title: "Topic Example Title",
        description: "Description Example description",
        createdAt: "Dummy date",
        createdBy: {
          id: 12,
          name: "Joe Doe",
          username: "exampleusername",
          downloadAvatar: "dsdasdasdasdasdadas.png",
        },
      },
    },
  },
});

const story = {
  id: 1,
  title: "Story Title",
  description: "Story description",
  totalLoves: 12,
  body: "<p>Simple Story</p>",
  createdBy: {
    id: 12,
    name: "Joe Doe",
    username: "exampleusername",
    downloadAvatar: "dsdasdasdasdasdadas.png",
  },
  topic: {
    id: 12,
    title: "Topic Example Title",
    description: "Description Example description",
    createdAt: "Dummy date",
    createdBy: {
      id: 12,
      name: "Joe Doe",
      username: "exampleusername",
      downloadAvatar: "dsdasdasdasdasdadas.png",
    },
  },
  authentication: {
    loggedIn: true,
  },
};

let mockMatch = {
  params: {
    id: 12,
  },
};

describe("Test for the EditStore Component", () => {
  let TestingComponent = () => (
    <MemoryRouter>
      <Provider store={store}>
        <EditStory match={mockMatch} />
      </Provider>
    </MemoryRouter>
  );

  let wrapper = mount(<TestingComponent />);

  test("should render component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should component be defined", () => {
    expect(EditStory).toBeDefined();
  });
});
