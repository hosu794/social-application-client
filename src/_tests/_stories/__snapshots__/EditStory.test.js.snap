// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the EditStore Component should render component correctly 1`] = `ReactWrapper {}`;
